snuba/clickhouse/optimize/optimize.py

```diff
@@ -22,17 +26,20 @@
 )
 from snuba.utils.metrics.wrapper import MetricsWrapper
 
-logger = logging.getLogger("snuba.optimize")
 
 # include thread and process info in log messages
-log_handler = logging.StreamHandler()
-log_handler.setFormatter(
-    logging.Formatter(
-        "%(levelname)s %(asctime)s %(name)s %(process)d-%(threadName)s %(message)s"
-    )
-)
-logger.addHandler(log_handler)
-logger.propagate = False
+def thread_info_processor(
```

Comment:
Why is this all necessary? Was this information automatically collected by the previous logger?

Reply:
Previous logger used the `logging.Formatter` class for format to add the thread info. `structlog` add this info by using the processor wrapper. We spin up threads for running the optimizer so we want visibility on thread id and thread name in all the logs.